### ISSUE TITLE
Investigate metrics configuration in Lightning

### ISSUE THREAD

**Issue by tizianocitro** [hypernetwork-research-group/hyperbench#90]
I want to investigate whether we should compute metrics only when doing `loss.backward()` using [Lightning features](https://lightning.ai/docs/torchmetrics/stable/pages/lightning.html).
